> Type check is still not fixed

This error is only in files this PR doesn't touch:
```
src/alchemlyb/parsing/lammps.py:659: error: "DataFrame" has no attribute "name"  [attr-defined]
src/alchemlyb/parsing/lammps.py:1015: error: "DataFrame" has no attribute "name"  [attr-defined]
src/alchemlyb/parsing/lammps.py:1118: error: "DataFrame" has no attribute "name"  [attr-defined]
src/alchemlyb/parsing/lammps.py:1314: error: "DataFrame" has no attribute "name"  [attr-defined]
src/alchemlyb/parsing/gomc.py:155: error: "DataFrame" has no attribute "name"  [attr-defined]
src/alchemlyb/parsing/gmx.py:251: error: "DataFrame" has no attribute "name"  [attr-defined]
Found 6 errors in 3 files (checked 31 source files)
```
Trying to fix it.

--- a/src/alchemlyb/parsing/gmx.py
+++ b/src/alchemlyb/parsing/gmx.py
@@ -248,7 +248,7 @@ def extract_dHdl(xvg: str, T: float, filter: bool = True) -> pd.DataFrame:
     newind = ["time"] + lambdas
     dHdl = dHdl.reset_index().set_index(newind)
 
-    dHdl.name = "dH/dl"
+    dHdl.name = "dH/dl"  # type: ignore[attr-defined]
 
     return dHdl
 

--- a/src/alchemlyb/parsing/gomc.py
+++ b/src/alchemlyb/parsing/gomc.py
@@ -152,7 +152,7 @@ def extract_dHdl(filename: str, T: float) -> pd.DataFrame:
     newind = ["time"] + cols
     dHdl = dHdl.reset_index().set_index(newind)
 
-    dHdl.name = "dH/dl"
+    dHdl.name = "dH/dl"  # type: ignore[attr-defined]
 
     return dHdl
 

--- a/src/alchemlyb/parsing/lammps.py
+++ b/src/alchemlyb/parsing/lammps.py
@@ -656,7 +656,7 @@ def extract_u_nk_from_u_n(
                     )
 
     u_nk.set_index(["time", "fep-lambda"], inplace=True)
-    u_nk.name = "u_nk"
+    u_nk.name = "u_nk"  # type: ignore[attr-defined]
 
     return u_nk
 
@@ -1012,7 +1012,7 @@ def extract_u_nk(
         u_nk.set_index(["time", "fep-lambda"], inplace=True)
     else:
         u_nk.set_index(["time", "coul-lambda", "vdw-lambda"], inplace=True)
-    u_nk.name = "u_nk"
+    u_nk.name = "u_nk"  # type: ignore[attr-defined]
 
     u_nk = u_nk.dropna()
 
@@ -1115,7 +1115,7 @@ def extract_dHdl_from_u_n(
 
     dHdl.set_index(["time", "fep-lambda"], inplace=True)
     dHdl["fep"] = dHdl["fep"] * beta
-    dHdl.name = "dH_dl"
+    dHdl.name = "dH_dl"  # type: ignore[attr-defined]
 
     return dHdl
 
@@ -1311,7 +1311,7 @@ def extract_dHdl(
         else:
             dHdl["coul"] = dHdl["coul"] * beta
 
-    dHdl.name = "dH_dl"
+    dHdl.name = "dH_dl"  # type: ignore[attr-defined]
 
     return dHdl
 

